### PR TITLE
Feature/slim 2184 datasource config

### PR DESCRIPTION
**Pull request by kroesctrl**
This PR contains 3 changes:
1. Made useProperties configurable for Scheduling, default values stay the same.
2. HibernateNamingStrategy.java the naming strategy lost the quotes parameter (this is needed, if a table or column name uses a privileged name like "user")
3. Added the timezone argument for the cron-expressions to the OsgpScheduler.java

### CODE REVIEW

File: osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/application/scheduling/OsgpScheduler.java [hunk 1]
```diff
@@ -130,24 +149,47 @@ public Trigger createJobTrigger(final JobDetail jobDetail, final String cronExpr
...
+     * @param cronExpression
+     *            The input for the trigger, a Quartz CRON expression like
+     *            {@code 0 0/1 * * * ?} for example.
+     *
+     * @throws SchedulerException
+     *             In case adding or scheduling of the job fails.
+     */
+    public void createAndScheduleJob(final Class<? extends Job> jobClass, final String cronExpression)
+            throws SchedulerException {
+        createAndScheduleJob(jobClass, cronExpression, TimeZone.getDefault());
```
Comment (smvdheijden) [inline osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/application/scheduling/OsgpScheduler.java:190]: Same as previous comment (and additionally consider using a constant for the UTC TimeZone):
```suggestion
        this.createAndScheduleJob(jobClass, cronExpression, TimeZone.getTimeZone("UTC"));
```
Consider defining a constant for the UTC TimeZone.

**Review by smvdheijden — CHANGES_REQUESTED** [review]

File: osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/application/scheduling/OsgpScheduler.java [hunk 2]
```diff
@@ -107,16 +108,34 @@ public JobDetail createJobDetail(final Class<? extends Job> jobClass) {
...
      */
-    public Trigger createJobTrigger(final JobDetail jobDetail, final String cronExpression) {
+    public Trigger createJobTrigger(final JobDetail jobDetail, final String cronExpression, final TimeZone timeZone) {
         return TriggerBuilder.newTrigger()
-                .forJob(jobDetail)
-                .withIdentity(jobDetail.getKey().getName() + "-Trigger")
-                .forJob(jobDetail)
-                .withSchedule(CronScheduleBuilder.cronSchedule(cronExpression))
-                .build();
+                             .forJob(jobDetail)
```
[the remark this answers is not in the document]
Comment (kroesctrl) [reply osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/application/scheduling/OsgpScheduler.java:118]: I reverted the indentation, i will try to look into this setting in the code formatting for Intelij

File: osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/application/scheduling/OsgpScheduler.java [hunk 3]
```diff
@@ -107,16 +108,34 @@ public JobDetail createJobDetail(final Class<? extends Job> jobClass) {
...
+     * @param jobDetail
+     *            A {@link JobDetail} instance, possibly created using
+     *            {@link OsgpScheduler#createJobDetail(Class)}.
+     * @param cronExpression
+     *            A CRON-expression.
+     *
+     * @return A {@link Trigger} instance.
+     */
+    public Trigger createJobTrigger(final JobDetail jobDetail, final String cronExpression) {
+        return createJobTrigger(jobDetail, cronExpression, TimeZone.getDefault());
```
[the remark this answers is not in the document]
Comment (kroesctrl) [reply osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/application/scheduling/OsgpScheduler.java:138]: I added this., but i tried to keep the code the same as it was before this change. TimeZone.getDefault() was used in the implementation of CronExpression
`    public TimeZone getTimeZone() {
        if (this.timeZone == null) {
            this.timeZone = TimeZone.getDefault();
        }

        return this.timeZone;
    }
`
i have no problems with changing it to TimeZone.getTimeZone("UTC") though

File: osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/application/scheduling/OsgpScheduler.java (hunk 1 above)
Comment (kroesctrl) [reply osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/application/scheduling/OsgpScheduler.java:190]: See comment above

Comment (kroesctrl) [reply osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/application/scheduling/OsgpScheduler.java]: Changed

File: osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/application/scheduling/OsgpScheduler.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (kroesctrl) [reply osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/application/scheduling/OsgpScheduler.java]: As discussed, changed to:
DateTimeZone.UTC.toTimeZone()

**Comment by smvdheijden** [comment]
Retest this please

**Review by smvdheijden — APPROVED** [review]

**Pull request merged**